File: Solutions/0860- Lemonade Change/Solution.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int fiveCounter = 0;
        int tenCounter = 0;

        for (int bill : bills) {
            if (bill == 5) {
                fiveCounter++;
            }
            else if (bill == 10) {
                if (fiveCounter > 0) {
                    fiveCounter--;
                    tenCounter++;
                } else {
                    return false;
                }
            } else {
                if (tenCounter > 0 && fiveCounter > 0) {
                    tenCounter--;
                    fiveCounter--;
                } else if (fiveCounter >= 3) {
                    fiveCounter -= 3;
                } else {
                    return false;
                }
            }
        }

        return true;
        
    }
}

File: Solutions/0860- Lemonade Change/Solution.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five_counter = 0
        ten_counter = 0

        for bill in bills:
            if bill == 5:
                five_counter += 1
            elif bill == 10:
                if five_counter > 0:
                    five_counter -= 1
                    ten_counter += 1
                else:
                    return False
            else:
                if ten_counter > 0 and five_counter > 0:
                    five_counter -= 1
                    ten_counter -= 1
                elif five_counter >= 3:
                    five_counter -= 3
                else:
                    return False
        return True

        

